jfix-stdlib-concurrency/src/test/java/ru/fix/stdlib/concurrency/futures/PendingFutureLimiterTest.java

```diff
@@ -282,6 +283,26 @@ public void waitAll_releases_thread_after_last_future_is_completed() throws Exce
         assertTimeoutPreemptively(assertingTaskTimeout, () -> limiter.waitAll(largeTimeout));
     }
 
+    @Test
+    public void enqueueBlocking_blocks_invoking_thread_when_limit_is_reached() throws Exception {
+        PendingFutureLimiter limiter = new LimiterBuilder()
+                .executionTimeLimit(0)
+                .enqueueTasks(3)
+                .build();
+        AtomicBoolean ai = new AtomicBoolean(false);
+        // We'll try to enqueue after futures timeout is exceeded
+        Executors.newSingleThreadScheduledExecutor().execute(() -> {
+            try {
+                limiter.enqueueBlocking(createTask());
+                ai.set(true);
+            } catch (InterruptedException ignore) {
+            }
+        });
+        assertFalse(ai.get());
+        Thread.sleep(TimeUnit.SECONDS.toMillis(10));
+        assertFalse(ai.get());
+    }
```

Comment:
We probably also should check that if we complete one of enqueued task, `enqueBlockingIsCompleteAndReturned` flag will became `true`

Reply:
Added check by releasing all tasks. Do you think it is necessary to add check by only one task releasing?

Reply:
Single task release should be sufficient to enable new one to being enqueued. So unleashLatchAndCompleteAllTask actually does not check exact behavior of the limiter: limiter should be able to enqueue in both cases: if you unleashAll or unleashSingle. Test should cover both cases or at least one where you are unleashing single task.

Reply:
Ok, changed test to single task releasing